packages/maskbook/src/utils/native-rpc/Web.ts

```diff
@@ -0,0 +1,27 @@
+import type { WebviewAPIs } from './types'
+import Services from '../../extension/service'
+import { definedSocialNetworkWorkers } from '../../social-network/worker'
+import * as settings from '../../settings/settings'
+import type { NativeAPISettingsName } from '../../settings/types'
+
+export const WebviewAPI: WebviewAPIs = {
+    web_echo: async (arg) => arg,
+    getDashboardURL: async () => browser.runtime.getURL(''),
+    getSettings: async (key: NativeAPISettingsName) => settings[key].value,
```

Comment:
You don't need to type the `key` here. Please remove type `NativeAPISettingsName`. It can be inferred from `WebviewAPIs` IMO.

Reply:
yes it is.

Reply:
No, the editor complains that key has an any type.

Reply:
yes it is.